src/operator/optimizer_op-inl.h

```diff
@@ -372,8 +372,8 @@ inline void SGDMomUpdateDnsRspDnsImpl(const SGDMomParam& param,
         auto mom_data = mom.dptr<DType>();
```

Comment:
nit: Can you use actual types for scalars? auto makes the code hard to read. Usually, it's most handy for long stuff like "std::unordered_map<std::string,  std::vector<int>>::const_iterator"